daos/post.go

```diff
@@ -0,0 +1,62 @@
+package daos
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"go-template/models"
+
+	"github.com/volatiletech/sqlboiler/v4/boil"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+)
+
+func FindPostbyId(id int, ctx context.Context) (*models.Post, error) {
+	contextExecutor := getContextExecutor(nil)
+	return models.Posts(qm.Where(fmt.Sprintf("%s=?", models.PostColumns.ID), id)).One(ctx, contextExecutor)
+}
+
+func FindPostbyTitle(title string, ctx context.Context) (*models.Post, error) {
+	contextExecutor := getContextExecutor(nil)
+	return models.Posts(qm.Where(fmt.Sprintf("%s=?", models.PostColumns.Title), title)).One(ctx, contextExecutor)
+}
+
+func FindPostByAuthor(authorId int, ctx context.Context) (models.PostSlice, int64, error) {
+	contextExecutor := getContextExecutor(nil)
+	posts, err := models.Posts(qm.Where(fmt.Sprintf("%s=?", models.PostColumns.AuthorID), authorId)).All(ctx, contextExecutor)
+	if err != nil {
+		return models.PostSlice{}, 0, err
+	}
+
+	count, err := models.Authors(qm.Where(fmt.Sprintf("%s=?", models.PostColumns.AuthorID), authorId)).Count(ctx, contextExecutor)
```

Comment:
If you are not using any qm here , or not paginating your query then you can simply return the slice length